File: b1002.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    // t : number of test case
    // x,y : axis
    // r : distance
    int t, x1, y1, r1, x2, y2, r2, result;
    // dis : distance between two turret
    double dis;

    //cout << "insert number of test case" << endl;
    cin >> t;

    while(t > 0) {
        //cout << "insert x1 y1 r1 x2 y2 r2" << endl;
        cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

        dis = sqrt(pow(x2-x1,2)+pow(y2-y1,2));

        // 원이 겹칠 때
        if(dis == 0 && r1 == r2) result = -1;
        // 원의 교점이 두개일 때
        // r1 + r2 가 거리보다 크면 교점이 두개
        // r2 - r1 의 절대값이 거리보다 적어도 두개
        else if(dis < r1 + r2 && abs(r2-r1) < dis) result = 2;
        // 원의 교점이 하나일 때
        // r1 + r2와 거리가 같다면 무조건 하나의 접점이 생김
        // 또한 
        else if(dis == r1 + r2) result = 1;
        else {
            result = 0;
        }

        cout << result << endl;

        t--;
    }

}

File: python/python_bj/b1002.py
# 터렛
from sys import stdin
import math

t = int(stdin.readline())
for i in range(t):
    x1,y1,r1,x2,y2,r2 = map(int, stdin.readline().split())

    distance = math.sqrt(pow(x2-x1,2)+pow(y2-y1,2))

    ## 원이 동심원일 때 혹은 일치
    if distance == 0 and r2 == r1 :
        print("-1")
    ## 원이 내접할 때 / 원이 외접할 때
    elif abs(r2-r1) == distance or (r2+r1) == distance :
        print("1")
    ## 원이 두 점에서 만날 때
    elif abs(r2-r1) < distance < (r2+r1)  :
        print("2")
    else :
        print("0")